src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -13,15 +13,34 @@
 using Microsoft.CodeAnalysis.SolutionCrawler;
```

Comment:
@dibarbet  for eyes on the changes in this file as well as you were heavily involved with the last refactoring here.